lib/utils.js

```diff
@@ -13,6 +13,25 @@ module.exports = {
     });
   },
 
+	getApiGatewayStageName() {
+		let stageNameStage, stageNameDeploy;
+		const serviceTemplate = this.serverless.service.resources;
+		Object.keys(serviceTemplate.Resources).forEach(function(key){
+			if (serviceTemplate.Resources[key]['Type'] == 'AWS::ApiGateway::Stage') {
+				stageNameStage = serviceTemplate.Resources[key].Properties.StageName;
```

Comment:
To determine if it is the one being deployed by ApiGatewayDeployement, should check the DeploymentId property of ApiGatewayStage is the ApiGatewayDeployement?